### PR TITLE
apis modules refactoring

### PR DESCRIPTION
**Pull request by Salaboy**
New APIs modules created for rest & services

### CODE REVIEW

**Comment by Salaboy** [comment]
@erdemedeiros @ryandawsonuk please review and merge

File: activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/ActivityCancelledEvent.java [hunk 1]
```diff
@@ -17,17 +17,17 @@
...
 package org.activiti.services.events;
 
-import org.activiti.services.core.model.events.ProcessEngineEvent;
+import org.activiti.services.api.events.ProcessEngineEvent;
 
 public interface ActivityCancelledEvent extends ProcessEngineEvent {
 
     /**
-     * @return the id of the activity this event is related to. This corresponds to an id defined in the process definition.
+     * @return the id of the activity this events is related to. This corresponds to an id defined in the process definition.
```
Comment (erdemedeiros) [inline activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/ActivityCancelledEvent.java]: Singular

File: activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/ActivityCancelledEvent.java [hunk 2]
```diff
@@ -17,17 +17,17 @@
...
      */
     String getActivityId();
 
     /**
-     * @return the name of the activity this event is related to.
+     * @return the name of the activity this events is related to.
```
Comment (erdemedeiros) [inline activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/ActivityCancelledEvent.java]: Singular

File: activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/ActivityCompensateEvent.java [hunk 3]
```diff
@@ -17,17 +17,17 @@
...
 package org.activiti.services.events;
 
-import org.activiti.services.core.model.events.ProcessEngineEvent;
+import org.activiti.services.api.events.ProcessEngineEvent;
 
 public interface ActivityCompensateEvent extends ProcessEngineEvent {
 
     /**
-     * @return the id of the activity this event is related to. This corresponds to an id defined in the process definition.
+     * @return the id of the activity this events is related to. This corresponds to an id defined in the process definition.
```
Comment (erdemedeiros) [inline activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/ActivityCompensateEvent.java]: Singular

File: activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/ActivityCompensateEvent.java [hunk 4]
```diff
@@ -17,17 +17,17 @@
...
      */
     String getActivityId();
 
     /**
-     * @return the name of the activity this event is related to.
+     * @return the name of the activity this events is related to.
```
Comment (erdemedeiros) [inline activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/ActivityCompensateEvent.java]: Singular

File: activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/ActivityCompletedEvent.java [hunk 5]
```diff
@@ -17,17 +17,17 @@
...
 package org.activiti.services.events;
 
-import org.activiti.services.core.model.events.ProcessEngineEvent;
+import org.activiti.services.api.events.ProcessEngineEvent;
 
 public interface ActivityCompletedEvent extends ProcessEngineEvent {
 
     /**
-     * @return the id of the activity this event is related to. This corresponds to an id defined in the process definition.
+     * @return the id of the activity this events is related to. This corresponds to an id defined in the process definition.
```
Comment (erdemedeiros) [inline activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/ActivityCompletedEvent.java]: Singular

File: activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/ActivityCompletedEvent.java [hunk 6]
```diff
@@ -17,17 +17,17 @@
...
      */
     String getActivityId();
 
     /**
-     * @return the name of the activity this event is related to.
+     * @return the name of the activity this events is related to.
```
Comment (erdemedeiros) [inline activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/ActivityCompletedEvent.java]: Singular

File: activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/ActivityStartedEvent.java [hunk 7]
```diff
@@ -17,17 +17,17 @@
...
 package org.activiti.services.events;
 
-import org.activiti.services.core.model.events.ProcessEngineEvent;
+import org.activiti.services.api.events.ProcessEngineEvent;
 
 public interface ActivityStartedEvent extends ProcessEngineEvent {
 
     /**
-     * @return the id of the activity this event is related to. This corresponds to an id defined in the process definition.
+     * @return the id of the activity this events is related to. This corresponds to an id defined in the process definition.
```
Comment (erdemedeiros) [inline activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/ActivityStartedEvent.java]: Singular

File: activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/ActivityStartedEvent.java [hunk 8]
```diff
@@ -17,17 +17,17 @@
...
      */
     String getActivityId();
 
     /**
-     * @return the name of the activity this event is related to.
+     * @return the name of the activity this events is related to.
```
Comment (erdemedeiros) [inline activiti-services/activiti-services-events/src/main/java/org/activiti/services/events/ActivityStartedEvent.java]: Singular

File: activiti-services/activiti-services-rest-api/.gitignore [hunk 9]
```diff
@@ -0,0 +1,13 @@
+/target
```
Comment (erdemedeiros) [inline activiti-services/activiti-services-rest-api/.gitignore]: Let's delete this file: all the content here is already ignored by the root .gitignore file.

File: activiti-services/activiti-services-rest-api/src/main/java/org/activiti/services/rest/api/TaskController.java [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+    @RequestMapping(value = "/{taskId}/claim", method = RequestMethod.POST)
+    Resource<Task> claimTask(@PathVariable String taskId);
+
+    @RequestMapping(value = "/{taskId}/release", method = RequestMethod.POST)
+    Resource<Task> releaseTask(@PathVariable String taskId);
+
+    @RequestMapping(value = "/{taskId}/complete", method = RequestMethod.POST)
+    ResponseEntity<Void> completeTask(@PathVariable String taskId,
+                                      @RequestBody(required = false) CompleteTaskCmd completeTaskCmd);
```
Comment (erdemedeiros) [inline activiti-services/activiti-services-rest-api/src/main/java/org/activiti/services/rest/api/TaskController.java]: Probably we should remove `required = false`

File: activiti-services/activiti-services-rest-api/src/main/java/org/activiti/services/rest/api/TaskController.java [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+    AuthenticationWrapper getAuthenticationWrapper();
+
+    void setAuthenticationWrapper(AuthenticationWrapper authenticationWrapper);
```
Comment (erdemedeiros) [inline activiti-services/activiti-services-rest-api/src/main/java/org/activiti/services/rest/api/TaskController.java]: Should this  be part of the interface? Isn't an implementation detail? If we keep it here we probably should add some java doc for it, as everything else is only exposing REST endpoints.

File: activiti-services/activiti-services-rest-impl/.gitignore [hunk 12]
```diff
@@ -0,0 +1,13 @@
+/target
```
Comment (erdemedeiros) [inline activiti-services/activiti-services-rest-impl/.gitignore]: Let's delete this file.

**Pull request merged**